core/qbft/qbft.go

```diff
@@ -261,9 +262,10 @@ func Run[I any, V comparable](ctx context.Context, d Definition[I, V], t Transpo
 			case uponQuorumCommits, uponJustifiedDecided: // Algorithm 2:8
 				// Applicable to any round (since can be justified)
 				round = msg.Round()
+				qCommit = justification
 
 				stopTimer()
-				qCommit = justification
+				timerChan = nil
```

Comment:
as per golang docs, stopping timers doesn't prevent them from firing. So explicit clearing of the timer is required.